option_test.go

```diff
@@ -413,3 +414,35 @@ func TestOptionValue(t *testing.T) {
 	is.EqualValues(None[string](), option2)
 	is.Nil(err2)
 }
+
+type SomeScanner struct {
+	Cool bool
+	Some int
+}
+
+func (ss *SomeScanner) Scan(src any) error {
+	val, ok := src.(string)
+	if !ok {
+		return fmt.Errorf("cannot scan - src is not a string")
+	}
+
+	var unmarshalled SomeScanner
+	if err := json.Unmarshal([]byte(val), &unmarshalled); err != nil {
+		return fmt.Errorf("failed to unmarshal json: %w", err)
+	}
+
+	*ss = unmarshalled
+	return nil
+}
+
+// If T is a sql.Scanner, make use of that.
+func TestOptionScanner(t *testing.T) {
+	is := assert.New(t)
+
+	jsonString := `{"cool": true, "some": 123}`
```

Comment:
Can you add a test for the case where optionalScanner would be of type `mo.None` please ?

Reply:
I added a case for when SQL `NULL` is scanned. Of course this is only as good as before and e.g. a JSON column which contains the string `null` will not result in `None` as there is no way to communicate that right now.